src/flamenco/runtime/fd_runtime.c

```diff
@@ -1335,29 +1335,49 @@ fd_runtime_finalize_txns_tpool( fd_exec_slot_ctx_t * slot_ctx,
         /* TODO: factor this out into a separate function */
         /* Even if a transaction fails, we still include the nonce account if it
            was part of the transaction. However, there is an exception to this:
-           if a recent blockhash is found then we do not include the nonce account. */
+           if a blockhash from the last 151 blockhashes in the blockhash queue
+           is found then we do not include the nonce account. */
```

Comment:
I would make it more clear that this is handled before transaction execution in agave

Reply:
and why we need to do it after the fact in firedancer